eth/protocols/eth/metrics.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2023 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package eth
+
+import "github.com/ethereum/go-ethereum/metrics"
+
+// meters stores ingress and egress handshake meters.
+var meters bidirectionalMeters
+
+// bidirectionalMeters stores ingress and egress handshake meters.
+type bidirectionalMeters struct {
+	ingress *hsMeters
+	egress  *hsMeters
+}
+
+// Get returns the corresponding meter depending if ingress or egress is
+// desired.
+func (h *bidirectionalMeters) Get(ingress bool) *hsMeters {
```

Comment:
Why is this method exported?

Reply:
All types in this file should be public!

Reply:
No they shouldn't. These are just for internal use.

Reply:
Always you and your 'internal API' nonsense. What if I want to use these in my app? It's useful code, why not allow me to use it too?

Reply:
It's not that much code honestly, you could just copy it to your own package.

Reply:
Always you with your internal API nonsense. It's useful code, why not allow me to use it too?

Reply:
It's not that much code though, I guess you could just copy it to your own package?

Reply:
A little copying is worse than a little dependency.

Reply:
Originally I was also calling this method in `eth/handler.go` to deal with the snap registration failure, but I ended up moving those meters to the snap package. No need to export it now! Sorry @anti-fjl.